src/core/bitop.d

```diff
@@ -501,19 +501,42 @@ struct BitRange
  */
 uint bswap(uint v) pure;
 
-/**
- * Swaps bytes in an 8 byte ulong end-to-end, i.e. byte 0 becomes
- * byte 7, byte 1 becomes byte 6, etc.
- */
-ulong bswap(ulong v) pure
-{
-    auto sv = Split64(v);
+version (CoreDdoc) {}
+else version (LDC) { /+ LDC treats bswap(ulong) as an intrinsic so don't change the mangling. +/ }
+else version (GNU) { /+ GDC treats bswap(ulong) as an intrinsic so don't change the mangling. +/ }
```

Comment:
Does it make sense to special-case these since LDC and GDC maintain their own versions of druntime anyway?

Reply:
I don't think so. You could also just use `pragma(inline, true)` to make it work for betterC instead of dummy-templatizing it.

Reply:
That works with `dmd -betterC -inline` but not if `-inline` is omitted (I just tested).

Reply:
Should be a DMD bug then, https://dlang.org/spec/pragma.html#inline:

> The default inline behavior is typically selectable with a compiler switch such as -inline.

Reply:
I don't think that's factually accurate.  GDC maintains its own implementation of `rt`, everything else is ad verbatim.

Reply:
> Does it make sense to special-case these since LDC and GDC maintain their own versions of druntime anyway?

Even if they maintain their own fork of druntime I think they prefer to have as few differences as possible.

Reply:
As far as I'm aware, the regression introduced in #2426 has never been fixed either (GDC treats all of core.checkedint as intrinsics, so mangling was broken there too).

Reply:
https://github.com/dlang/druntime/pull/3162